server/src/main/java/org/elasticsearch/action/admin/indices/validate/query/QueryExplanation.java

```diff
@@ -88,7 +92,11 @@ public void readFrom(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeString(index);
+        if (out.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            out.writeOptionalString(index);
+        } else {
+            out.writeString(index);
+        }
```

Comment:
I think that we should back port this fix to 6.4. These version checks are fine to leave here for now but when you back port you will need to change them to `Version.V_6_4_0` in the back ported commit on the 6.x branch an then after that add a new commit to the master branch to change these checks to `Version.V_6_4_0` on master too. This process avoids getting CI failures before you've had a chance to back port the commit to 6.x.